### PR TITLE
add method to schedule thoth-repo-init workflow

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

https://github.com/thoth-station/thoth-application/pull/1965

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

For scheduling new workflow

### CODE REVIEW

File: thoth/common/workflows.py [hunk 1]
```diff
@@ -885,3 +885,25 @@ def submit_graph_sync(
...
+    def submit_thoth_repo_init(
+        self,
+        template_parameters: Optional[Dict[str, str]] = None,
+        workflow_parameters: Optional[Dict[str, Any]] = None,
+    ) -> Optional[str]:
+        """Submit repo initialization workflow."""
+        if not self.openshift.infra_namespace:
+            raise ConfigurationError("Infra namespace was not provided.")
+
+        if not self.openshift.middletier_namespace:
```
Comment (harshad16) [inline thoth/common/workflows.py:898]: As this is serving to Users, should this execute in the backend namespace?

**Review by harshad16 — CHANGES_REQUESTED** [review]
one doubt

File: thoth/common/workflows.py (hunk 1 above)
Comment (fridex) [reply thoth/common/workflows.py:898]: +1

File: thoth/common/workflows.py [hunk 2]
```diff
@@ -885,3 +885,25 @@ def submit_graph_sync(
...
+
+        if not self.openshift.infra_namespace:
+            raise ConfigurationError("Infra namespace was not provided.")
+
+        workflow_id: Optional[str] = self.submit_workflow_from_template(
+            self.openshift.backend_namespace,
+            label_selector="template=thoth-repo-init",
+            template_parameters=template_parameters or {},
+            workflow_parameters=workflow_parameters or {},
+            workflow_namespace=self.openshift.middletier_namespace,
```
Comment (pacospace) [inline thoth/common/workflows.py:906]: `openshift.backend_namespace` where the workflow is going to be scheduled

File: thoth/common/workflows.py [hunk 3]
```diff
@@ -885,3 +885,25 @@ def submit_graph_sync(
...
+    ) -> Optional[str]:
+        """Submit repo initialization workflow."""
+        if not self.openshift.backend_namespace:
+            raise ConfigurationError("Backend namespace was not provided.")
+
+        if not self.openshift.infra_namespace:
+            raise ConfigurationError("Infra namespace was not provided.")
+
+        workflow_id: Optional[str] = self.submit_workflow_from_template(
+            self.openshift.backend_namespace,
```
Comment (pacospace) [inline thoth/common/workflows.py:902]: `self.openshift.infra_namespace` where the workflow template is located

**Review by pacospace — CHANGES_REQUESTED** [review]

**Review by fridex — APPROVED** [review]
lgtm 👍🏻

**Pull request merged**